src/middlewares/Fetch.ts

```diff
@@ -1,22 +1,26 @@
 import {combineUrlWithBaseUrl, combineUrlWithQueryParameters} from "../internal/formatting";
 import {IMiddleware} from "../Stack";
+import {IFetchRequestCacheOptions} from "./Cache";
 
 export type FetchHeaders = {
     [key: string]: string;
+    forEach?: any;
 };
 
 export type FetchResponse<T> = {
     new(body: any, init: any): FetchResponse<T>;
     body: string;
     bodyUsed: boolean;
-    headers?: FetchHeaders;
+    headers: FetchHeaders;
     ok: boolean;
     status: number;
     statusText: string;
     type: string;
     url: string;
+    cache?: { used: boolean; timestamp: number; }; // signifies that response has been reconstructed from cache
```

Comment:
actually `cache` is not the official spec for [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) but we have to flag it (our spec). it means we can use `IFetchResponseCacheOptions`